controllers/kserve_customcacert_controller.go

```diff
@@ -0,0 +1,190 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/opendatahub-io/odh-model-controller/controllers/constants"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	kserveCustomCACertConfigMapName = constants.KServeCACertConfigMapName
+	odhGlobalCACertConfigMapName    = constants.ODHGlobalCertConfigMapName
+)
+
+type KServeCustomCACertReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+}
+
+// reconcileConfigMap watch odh global ca cert and it will create/update/delete kserve custom cert configmap
+func (r *KServeCustomCACertReconciler) reconcileConfigMap(configmap *corev1.ConfigMap, ctx context.Context) error {
+	// Initialize logger format
+	log := r.Log
+
+	odhCustomCertData := configmap.Data[constants.ODHCustomCACertFileName]
+	if odhCustomCertData == "" {
+		log.Info(fmt.Sprintf("Detected opendatahub global cert ConfigMap (%s), but custom cert is not set\n", odhGlobalCACertConfigMapName))
+		kserveCustomCertConfigMap := &corev1.ConfigMap{}
+		err := r.Get(ctx, types.NamespacedName{Name: kserveCustomCACertConfigMapName, Namespace: configmap.Namespace}, kserveCustomCertConfigMap)
+		if err == nil {
+			log.Info(fmt.Sprintf("Deleting KServe custom cert ConfigMap (%s)", kserveCustomCACertConfigMapName))
+			if err := r.Delete(ctx, kserveCustomCertConfigMap); err != nil {
+				return err
+			}
+		} else if !apierrs.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
+
+	configData := map[string]string{constants.KServeCACertFileName: odhCustomCertData}
+	newCaCertConfigMap := getDesiredCaCertConfigMapForKServe(kserveCustomCACertConfigMapName, configmap.Namespace, configData)
+
+	kserveCustomCertConfigMap := &corev1.ConfigMap{}
+	if err := r.Get(ctx, types.NamespacedName{Name: kserveCustomCACertConfigMapName, Namespace: configmap.Namespace}, kserveCustomCertConfigMap); err != nil {
+		if !apierrs.IsNotFound(err) {
+			return err
+		}
+		log.Info(fmt.Sprintf("Creating KServe custom cert ConfigMap (%s) because it detected the creation of the opendatahub global cert ConfigMap (%s)", kserveCustomCACertConfigMapName, odhGlobalCACertConfigMapName))
+		if err := r.Create(ctx, newCaCertConfigMap); err != nil {
```

Comment:
If lifecycle of `kserveCustomCACertConfigMap` depends on `odhGlobalCACertConfigMap` then could we define the parent child relationship between them.

Reply:
odhGlobalCACertConfigMap will be created by opendtahub operator and odh-model-controller reply on platform operator.